### PR TITLE
Fix unused DatasetInfosDict code in push_to_hub

### PR DESCRIPTION
**Pull request by lhoestq**

### CODE REVIEW

File: src/datasets/dataset_dict.py [hunk 1]
```diff
@@ -1722,8 +1711,15 @@ def push_to_hub(
                 MetadataConfigs({"default": default_metadata_configs_to_dump}).to_dataset_card_data(dataset_card_data)
         # push to the deprecated dataset_infos.json
         if config.DATASETDICT_INFOS_FILENAME in repo_files:
+            download_config = DownloadConfig()
```
Comment (polinaeterna) [inline src/datasets/dataset_dict.py:1714]: why did you add this here?

File: src/datasets/dataset_dict.py [hunk 2]
```diff
@@ -1722,8 +1711,15 @@ def push_to_hub(
...
+            download_config.download_desc = "Downloading metadata"
+            download_config.token = token
+            dataset_infos_path = cached_path(
+                hf_hub_url(repo_id, config.DATASETDICT_INFOS_FILENAME),
+                download_config=download_config,
+            )
```
Comment (lhoestq) [inline src/datasets/dataset_dict.py:1720]: this code needs to be here, because in the previous location it would be run only if there is no README.md and if there is a json file. But it needs to run if there is a json file, no matter if there is a README.md or not

Comment (polinaeterna) [reply src/datasets/dataset_dict.py:1720]: ah lol true good catch!

**Review by polinaeterna — COMMENTED** [review]
thank you :heart:

**Review by polinaeterna — APPROVED** [review]
